### PR TITLE
Suggest string messages

### PR DESCRIPTION
**Pull request by SwissalpS**
@S-S-X is this what you meant by adding both string and table messages in a clean way?

### CODE REVIEW

**Comment by S-S-X** [comment]
~not exactly, my suggestion was simpler API and -5 LOC not +16 LOC~

edit. and actually this was not part of code I talked about in chat, did not read other parts at all... only those few lines that caused crash. So above was actually for another commit.

Maybe it was someone else suggesting string messages. But yes would still agree that strings are useful in some situations as that allows using digiline buttons or keyboard etc. to send messages directly to nodes without using lua controller.

File: rgb.lua [hunk 1]
```diff
@@ -13,10 +23,29 @@ local digiline_rules = {
...
+    local s = mixed_msg:lower()
+    if 'on' == s then
+      msg.switch = 1
+    elseif 'off' == s then
+      msg.switch = 0
+    elseif 'reset' == s then
+      msg.reset = 1
+    --elseif '#' == s:sub(1, 1) then
+      -- TODO:
+      --msg.color = translatedColour(s)
```
Comment (S-S-X) [inline rgb.lua:34]: this is easy to implement with `tonumber`, just feed it splitted hex values to get r, g and b.
```lua
s = "#a1b2c3" -- dec 161, 178, 195
r = tonumber("0x" .. s:sub(2,3))
g = tonumber("0x" .. s:sub(4,5))
b = tonumber("0x" .. s:sub(6,7))
```

**Comment by SwissalpS** [comment]
> buttons or keyboard etc. to send messages directly to nodes without using lua controller

exactly :)

File: rgb.lua (hunk 1 above)
Comment (SwissalpS) [reply rgb.lua:34]: I didn't write this part, because Grid was already working on that. Thanks for the snippet anyway.

**Review by 6r1d — APPROVED** [review]
Thanks.

File: rgb.lua (hunk 1 above)
Comment (6r1d) [reply rgb.lua:34]: @S-S-X, I think it's not that simple. 256-color palette, used by unifieddyes seem to be having HSV structure. It has 24 hues, 10 levels of value and I'm not sure how and if it uses saturation. The closest approach I see there is to convert the color like you do to find a hue, apply value and... Do something about the saturation level, as well.

**Pull request merged**